scripts/perftest/genALSData.sh

```diff
@@ -21,30 +21,31 @@
 #-------------------------------------------------------------
 
 CMD=$1
-DATADIR=$2
+DATADIR=$2/als
 
-FORMAT="binary" # can be csv, mm, text, binary
+FORMAT="text" # can be csv, mm, text, binary
 DENSE_SP=0.9
 SPARSE_SP=0.01
 
-BASEPATH=$(dirname $0)
+EXTRADOT=""
+if [ "${CMD}" == "systemds" ]; then EXTRADOT="." ; fi
 
 #generate XS scenarios (80MB)
-${CMD} -f ${BASEPATH}/../datagen/genRandData4ALS.dml --nvargs X=${DATADIR}/X10k_1k_dense rows=10000 cols=1000 rank=10 nnz=`echo "scale=0; 10000 * 1000 * $DENSE_SP" | bc` sigma=0.01 fmt=$FORMAT
-${CMD} -f ${BASEPATH}/../datagen/genRandData4ALS.dml --nvargs X=${DATADIR}/X10k_1k_sparse rows=10000 cols=1000 rank=10 nnz=`echo "scale=0; 10000 * 1000 * $SPARSE_SP" | bc` sigma=0.01 fmt=$FORMAT
+${CMD} -f ${EXTRADOT}./datagen/genRandData4ALS.dml --nvargs X=${DATADIR}/X10k_1k_dense rows=10000 cols=1000 rank=10 nnz=`echo "scale=0; 10000 * 1000 * $DENSE_SP" | bc` sigma=0.01 fmt=$FORMAT
```

Comment:
I'm pretty sure that this change is not needed, since the script is intended to be run one directory further in,
I would suggest that you add a comment int the top specifying where to run it from, and even better an error that says, "wrong directory, please cd to xyz"

Reply:
Do you mean one folder further in, in the fed folder?
I've changed that script to be used by the general ALS benchmarks, as it was previously used by the federated ALS benchmark. The federated benchmarks now have a new genALS_FedData.sh in the fed folder.